pkg/tfbridge/detailed_diff.go

```diff
@@ -149,6 +152,37 @@ func (differ detailedDiffer) getEffectiveType(path walk.SchemaPath) shim.ValueTy
 	return tfs.Type()
 }
 
+type hashIndexMap map[int]int
+
+func (differ detailedDiffer) calculateSetHashIndexMap(path propertyPath, listVal resource.PropertyValue) hashIndexMap {
+	identities := make(hashIndexMap)
+
+	tfs, ps, err := lookupSchemas(path, differ.tfs, differ.ps)
+	if err != nil {
+		return nil
+	}
+
+	convertedVal, err := makeSingleTerraformInput(context.Background(), path.String(), listVal, tfs, ps)
+	if err != nil {
+		return nil
+	}
+
+	if convertedVal == nil {
+		return nil
+	}
+
+	convertedListVal, ok := convertedVal.([]interface{})
+	contract.Assertf(ok, "converted value should be a list")
```

Comment:
Collection itself being unknown or nil does not trigger this panic right?

Reply:
No, we've already checked that this takes a list. Note the signature of the function was changed to make this clearer.